### ISSUE TITLE
Include backup substitution for Google backup

### ISSUE THREAD

**Issue by rugk** [lineageos4microg/l4m_website#35]
There is a new great app for backing up the _whole_ Android system in development by @grote mostly:

https://github.com/stevesoltys/backup  

But first, have a look on **how it works**:
https://twitter.com/t_grote/status/1175070283247628291
https://twitter.com/t_grote/status/1174022555612647426

It works without root.
And can backup encrypted backups to several locations, even using Nextcloud for backups. 

---

Now there is a problem: As it uses internal APIs, it does _not only_ need to be installed as a system-app, but the app _also_ needs to be signed by the platforms private key. See https://github.com/stevesoltys/backup/issues/8#issuecomment-515891338 for that statement.

Thus, installing afterwards on Android is not possible.

As the app basically replaces _Google's cloud backup_ and the aim of this project/microG is to replace all built-in Google apps with free ones, IMHO, this fits into this project very well.

So final question: Could not you include this in your builds?

**Comment by grote** [comment]
Ideally, this would be included in LineageOS upstream directly.

**Comment by rugk** [comment]
I doubt they do that. They even refuse to make their OS compatible with microG (or even ship a microG build), because they would have to enable signature spoofing.
If LineageOS could, I guess, they would ship with Google apps.

**Comment by rugk** [comment]
Well… they do not even allow one to submit feature requests, anywhere… lol…

Even on reddit this is their guideline:
![image](https://user-images.githubusercontent.com/11966684/65386401-9094f180-dd3b-11e9-9cc8-fb9065e91de8.png)

---

I ignored this and created this post:
https://www.reddit.com/r/LineageOS/comments/d7p5yc/backup_solution_replacement_for_google_backup_in/

**Comment by hex-m** [comment]
That means we need to [send a patch](https://wiki.lineageos.org/submitting-patch-howto.html) to start the discussion and find out if they are interested. It's basically "proof of work" as a spam avoidance strategy.

**Comment by grote** [comment]
Before sending a patch, you would need to test how the Backup app behaves with flashed gapps. Since LineageOS supports (and probably wants to continue to support) flashing gapps, a system backup app should not interfere with the gapps backup app.

Test reports can be send here: https://github.com/stevesoltys/backup/issues/new

**Comment by hex-m** [comment]
Actually, LineageOS does _not_ support gapps.

> The Google apps packages are not supported in any way by LineageOS

https://wiki.lineageos.org/gapps.html

But if the patch breaks all installations that use gapps, it may still be seen as a problem. ;)

**Comment by rugk** [comment]
Of course they see it as a problem. And they totally support it, in the sense, that they make it very clear and easy (in installation guides etc.) in how to install GApps in LineageOS.

**Comment by rugk** [comment]
BTW the project has been renamed to "seedvault" and [v1.0.0](https://github.com/stevesoltys/seedvault/releases/tag/1.0.0) has been released. :tada:

**Comment by Moini** [comment]
Has anyone been able to start working on this?

I've found out today that I'm not going to get any updates for my LineageOS 16 system (they say it would overload their build servers to continue providing builds for 16 when 17 is supported on a device), and knowing that I'm not going to be able to make backups on LOS 17 is a no-go as well.

Unfortunately, I can't actually help, but can only ask for status... :-/

**Comment by ArchangeGabriel** [comment]
Note that SeedVault has finally been implemented in upstream LOS, and is thus now available in this project too.

**Comment by rugk** [comment]
Yeah, so I guess we can close this issue here… :slightly_smiling_face:
